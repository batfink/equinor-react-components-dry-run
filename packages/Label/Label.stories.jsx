import React from 'react';
import { storiesOf } from '@storybook/react';
import styled, { createGlobalStyle } from 'styled-components';
import { energy, moss } from '../themes/colors';
import Label from './lib/Label';

const stories = storiesOf('Label', module);

/* eslint-disable no-unused-expressions */
const GlobalStyle = createGlobalStyle`
    .energy {
        color: ${energy};
    }
`;

const Wrapper = styled.div`
    display: grid;
    align-items: center;
    justify-items: center;
    height: 100vh;
`;

stories.addDecorator(story => (
    <Wrapper>
        <GlobalStyle />
        {story()}
    </Wrapper>
));

stories
    .add('Default', () => <Label>Some label</Label>)
    .add('With className', () => <Label className="energy">Some label</Label>)
    .add('With style', () => <Label style={{ color: moss }}>Some label</Label>);
