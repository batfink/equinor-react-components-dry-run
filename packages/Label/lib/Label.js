import styled from 'styled-components';
import { mist } from '../../themes/colors';

const Label = styled.label`
    display: grid;
    grid-gap: 0.25rem;
    color: ${mist};
    font-weight: 500;
    font-size: 1.125rem;
`;

export default Label;
