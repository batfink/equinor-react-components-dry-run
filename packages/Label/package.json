{
  "name": "label",
  "main": "Label",
  "version": "0.0.1",
  "dependencies": {
    "prop-types": "^15.6.2",
    "react": "^16.5.2",
    "styled-components": "^4.0.0-beta.10"
  },
  "description": "Label",
  "keywords": [
    "Label",
    "html"
  ],
  "author": "fortena <arnigeir@gmail.com>",
  "homepage": "https://github.com/batfink/equinor-react-components-dry-run#readme",
  "license": "ISC",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/batfink/equinor-react-components-dry-run.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/batfink/equinor-react-components-dry-run/issues"
  }
}
