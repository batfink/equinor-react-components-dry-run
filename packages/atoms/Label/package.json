{
    "name": "@equinor/react-label",
    "main": "dist/Label.js",
    "module": "src/Label.jsx",
    "version": "0.0.0",
    "license": "UNLICENSED",
    "scripts": {
        "prepare": "yarn build",
        "clean": "rimraf dist && mkdir dist",
        "bundle": "rollup -c",
        "build": "npm-run-all clean bundle"
    },
    "peerDependencies": {
        "react": "^16.0.0",
        "react-dom": "^16.0.0",
        "prop-types": "^15.6.2",
        "styled-components": "^4.0.0-beta"
    }
}
