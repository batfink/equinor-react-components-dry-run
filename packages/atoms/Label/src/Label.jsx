import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { mist } from './colors'

const Label = styled.label`
    display: grid;
    grid-gap: 0.25rem;
    color: ${mist};
    font-weight: 500;
    font-size: 1.125rem;
 `

export default Label
