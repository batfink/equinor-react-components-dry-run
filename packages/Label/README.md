# `label`

Label extends the label html element

## Usage

```
import Label from '@equinor-ui-components/Label';

<Label>Name</Label>
```
