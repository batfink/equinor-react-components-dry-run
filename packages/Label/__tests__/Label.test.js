import React from 'react';
import Label from '../lib/Label';
import renderer from 'react-test-renderer';

describe('Label', () => {
    it('should render a Label', () => {
        const tree = renderer.create(<Label>Label</Label>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
