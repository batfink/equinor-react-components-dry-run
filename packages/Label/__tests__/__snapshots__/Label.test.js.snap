// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render a Label 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.25rem;
  color: #d7ebf4;
  font-weight: 500;
  font-size: 1.125rem;
}

<label
  className="c0"
>
  Label
</label>
`;
